test/uri/uri_canonicalize_test.cc

```diff
@@ -68,10 +68,16 @@ TEST(URI_canonicalize, example_10) {
   EXPECT_EQ(uri.recompose(), "urn:example:schema");
 }
 
-TEST(URI_canonicalize, example_11) {
+TEST(URI_canonicalize, example_relative_1) {
   sourcemeta::jsontoolkit::URI uri{"../foo"};
   uri.canonicalize();
-  EXPECT_EQ(uri.recompose(), "/foo");
+  EXPECT_EQ(uri.recompose(), "foo");
```

Comment:
⚠️ @jviotti

Reply:
I think this one is OK.

Reply:
This one should be fixed too. I think this indicates that storing path components in a vector without the slash might not do, as we might lose some information? Maybe the path components here should look like this (i.e. keeping the slash as part of the component?):

```c++
[ "/foo" ]
```

Reply:
I don't know so far. I'll think about it.

But if add the slash in `recompose` then we will end up having `/example.com/foo` for the input `example.com/foo`.

Reply:
Now the thing is that we use `uriparser` for the parsing, and we won't have the information if the slash was there or not. We could check `this->data` to see if the there is slash or not before the path.

Reply:
Yeah, sounds tricky. The thing is that if we get a URI and recompose, it should ideally get you back the exact same thing you put in there, and any updates to clean it up, etc should only happen under canonicalisation.

Maybe keeping the path component as a vector doesn't help much?

Reply:
> Maybe keeping the path component as a vector doesn't help much?

yeah we should keep a string a server a span on demand!